EMMItem.cs

```diff
@@ -84,11 +164,13 @@ public override bool OnPickup(Item item, Player player)
 			ModifierContext ctx = new ModifierContext { Method = ModifierContextMethod.OnPickup, Item = item, Player = player };
 
 			Modifier m = GetModifier(item);
-			if (m == null)
-				GetItemInfo(item)?.RollNewModifier(ctx);
+			if (!hasRolled && m == null)
```

Comment:
I see you've incorporated hasRolled here in OnPickup, but where is it for OnCraft?

Reply:
There shouldn't be a need for it to be in OnCraft, as it triggers when the item is crafted. Therefore every time OnCraft is called it should be on a new item with hasRolled set to false.

Reply:
Good observation, however it's still worthwhile to add. Modders can pull finicky code and still trigger something you may not have considered (i.e. OnCraft) Consider it safeguarding, it might not happen in the regular use case but it may in another you hadn't conceived

Reply:
On another note, we will need to add additional checks in save and load, because now we've included hasRolled and it can save if it has rolled. In _Save we must check if Modifier is not null, if it is null we should put the Modifier as null in the TC. (otherwise we do regular saving) In _Load, we must check if the Modifier key in TC is null, if not then we can load it.